mico-core/src/main/java/io/github/ust/mico/core/service/MicoStatusService.java

```diff
@@ -33,95 +57,151 @@
                 micoService.getShortName(), micoService.getVersion(), e.getMessage());
+            return serviceStatus.setErrorMessages(CollectionUtils.listOf("Error while retrieving Kubernetes deployment of MicoService " + micoService.getShortName() + " "
+                + micoService.getVersion() + " .  Caused by: " + e.getMessage()));
         }
-        if (!deploymentOptional.isPresent()) {
-            log.warn("There is no deployment of the MicoService '{}' '{}'. Continue with next one.",
-                micoService.getShortName(), micoService.getVersion());
-        }
+        serviceStatus
+            .setName(micoService.getName())
+            .setShortName(micoService.getShortName())
+            .setVersion(micoService.getVersion());
 
-        Deployment deployment = deploymentOptional.get();
-        MicoServiceStatusDTO serviceStatus = new MicoServiceStatusDTO();
-        serviceStatus.setName(micoService.getName());
-        serviceStatus.setShortName(micoService.getShortName());
-        serviceStatus.setVersion(micoService.getVersion());
-        serviceStatus.setAvailableReplicas(deployment.getStatus().getAvailableReplicas());
-        serviceStatus.setRequestedReplicas(deployment.getSpec().getReplicas());
+        // Get status information for service interfaces of a service
+        serviceStatus.setInterfacesInformation(getServiceInterfaceStatus(micoService));
 
-        // Get status information for service interfaces of a serivce
-        List<MicoServiceInterfaceDTO> micoServiceInterfaceDTOList = getServiceInterfaceStatus(micoService);
-        serviceStatus.setInterfacesInformation(micoServiceInterfaceDTOList);
+        // Return the names of all applications that are using this service
+        List<MicoApplication> usingApplications = micoApplicationRepository.findAllByUsedService(micoService.getShortName(), micoService.getVersion());
+        for (MicoApplication micoApplication : usingApplications) {
+            BasicMicoApplicationDTO usingApplication = new BasicMicoApplicationDTO(micoApplication.getShortName(), micoApplication.getVersion(), micoApplication.getName());
+            serviceStatus.getUsingApplications().add(usingApplication);
+        }
 
         // Get status information for all pods of a service
         List<Pod> podList = micoKubernetesClient.getPodsCreatedByDeploymentOfMicoService(micoService);
-        List<KubernetesPodInfoDTO> podInfos = new LinkedList<>();
+        List<KubernetesPodInformationDTO> podInfos = new ArrayList<>();
+        // Get all the nodes on which the pods of a deployment of a MicoService are running
+        Map<String, List<Pod>> podsPerNode = new HashMap<>();
         for (Pod pod : podList) {
-            KubernetesPodInfoDTO podInfo = getUiPodInfo(pod);
-            podInfos.add(podInfo);
+            if (podsPerNode.containsKey(pod.getSpec().getNodeName())) {
+                podsPerNode.computeIfPresent(pod.getSpec().getNodeName(), (s, pods) -> {
```

Comment:
This code part is currently not covered by an unit test in `MicoStatusServiceTest`. Consider to add one.

Reply:
I added a scenario with four pods running on two different nodes for one service to the tests.